iohub/reader_base.py

```diff
@@ -1,32 +1,111 @@
-import numpy as np
 import zarr
+from numpy.typing import DTypeLike, NDArray
 
 
 class ReaderBase:
     def __init__(self):
-        self.frames = None
-        self.channels = None
-        self.slices = None
-        self.height = None
-        self.width = None
-        self.dtype = None
-        self.mm_meta = None
-        self.stage_positions = None
-        self.z_step_size = None
-        self.channel_names = None
+        self.frames: int = None
+        self.channels: int = None
+        self.slices: int = None
+        self.height: int = None
+        self.width: int = None
+        self.dtype: DTypeLike = None
+        self._mm_meta: dict = None
+        self._stage_positions: list[dict] = []
+        self.z_step_size: float = None
+        self.channel_names: list[str] = None
 
     @property
     def shape(self):
+        """Get the underlying data shape as a tuple.
+
+        Returns
+        -------
+        tuple
+            (frames, slices, channels, height, width)
+
+        """
         return self.frames, self.channels, self.slices, self.height, self.width
 
-    def get_zarr(self, position: int) -> zarr.array:
-        pass
+    @property
+    def mm_meta(self):
+        return self._mm_meta
 
-    def get_array(self, position: int) -> np.ndarray:
-        pass
+    @mm_meta.setter
+    def mm_meta(self, value):
+        if not isinstance(value, dict):
+            raise TypeError(
+                f"Type of `mm_meta` should be `dict`, got `{type(value)}`."
+            )
+        self._mm_meta = value
```

Comment:
Why does the base reader needs `mm_` properties? Shouldn't this be in the `Micromanager` classes?

Reply:
> How is it saved in the zarr case?

It's a field of zattrs

Reply:
What do you think about changing this behavior in this PR or another PR?
In my opinion, the codebase would improve by not having this method in every child, and the functionality could be added with a Mixin, because the `_set_mm_data` looks very similar for every class.

Reply:
Both would work. Probably in another PR though so we can merge this one faster. API won't change with a Mixin since currently every child has the method.

Reply:
Sounds good!

Reply:
Created #85